### ISSUE TITLE
How to decouple workspace platform from project platform?

### ISSUE THREAD

**Issue by Dragnalith** [premake/premake-core#1243]
TL;DR: In visual studio, when a solution platform is selected, I want project being able to be built for another platform, or not being built at all.

I am working for a video game studio with big cross-compilation need.

I have one big solution to manage the build of the entire project. It includes different kind of projects
- Runtime library and processes which are cross-platform projects. (let's say Windows and Playstation)
- Code generator tools (Windows only)
- Asset Editor (Windows only)
- Samples (can be Windows only or Playstation only)

When I compile for Playstation, I select Playstation as my solution platform. In that case my runtime module needs to be compiled for playstation, but my code generator tools and asset editor tools needs to be compiled for windows. In addition I would like to disable the build of some project depending on the platform.

I know how to do it with visual studio. You can configure everything in _Configuration Manager_ of the solution. But I do not know if it is possible to do it with premake and if possible how to do it.

I am used to customize premake but in that case I am concern about the platform/configuration data model of premake. Since there is not difference between workspace and project platform, I am not sure if this feature is possible to exist when current premake design.

**Comment by samsinsane** [comment]
It sounds like you're looking for [`configmap`](https://github.com/premake/premake-core/wiki/configmap).

**Comment by Dragnalith** [comment]
@samsinsane yes, this is roughly good enough for my use case. Thank you! :-)

Btw, I have discovered a cumbersome point. Since platform property is inherited from workspace to project, there is no way to restrict platforms of a project (e.g Windows only project) without using `removeplatforms` property. But that mean I have to maintain in a global the list of platform used by the workspace to give them as argument of `removeplatforms` to all the windows only project.

I am not sure if it is a current premake limitation, or if I am missing some knowledge.

**Comment by samsinsane** [comment]
That is documented at the end of [this](https://github.com/premake/premake-core/wiki/Configurations-and-Platforms) page in the wiki. There's two ways you can get around this:
```lua
-- Always build Windows, I'm fairly certain this works
configmap { ["*"] = "Windows" }

-- Or, only build when Windows
removeplatforms "*"
platforms "Windows"
```

The key thing is that you can't _add_ new `platforms` to the workspace, but that doesn't stop you from adding `platforms` to the project and mapping those to workspace `platforms`. There's an example of that in our unit tests here: https://github.com/premake/premake-core/blob/ba3b2eb50b70eb0d85c55512b9cd3814a8237a71/tests/project/test_config_maps.lua#L109-L121

The `workspace` is defined further up here:https://github.com/premake/premake-core/blob/ba3b2eb50b70eb0d85c55512b9cd3814a8237a71/tests/project/test_config_maps.lua#L16-L19

**Comment by Dragnalith** [comment]
Thank you, that solve all my issues!
